packages/salesforcedx-vscode-core/src/commands/templates/forceApexClassCreate.ts

```diff
@@ -20,41 +18,41 @@ import {
   SfdxWorkspaceChecker
 } from '../util';
 import { OverwriteComponentPrompt } from '../util/postconditionCheckers';
-import { LibraryBaseTemplateCommand } from './libraryBaseTemplateCommand';
-import { APEX_CLASS_DIRECTORY, APEX_CLASS_TYPE } from './metadataTypeConstants';
+import { LibraryForceApexClassCreateExecutor } from './executors/LibraryForceApexClassCreateExecutor';
+import {
+  APEX_CLASS_DIRECTORY,
+  APEX_CLASS_NAME_MAX_LENGTH,
+  APEX_CLASS_TYPE
+} from './metadataTypeConstants';
 
-export class LibraryForceApexClassCreateExecutor extends LibraryBaseTemplateCommand<
-  DirFileNameSelection
-> {
-  public executionName = nls.localize('force_apex_class_create_text');
-  public telemetryName = 'force_apex_class_create';
-  public metadataTypeName = APEX_CLASS_TYPE;
-  public templateType = TemplateType.ApexClass;
-  public getOutputFileName(data: DirFileNameSelection) {
-    return data.fileName;
-  }
-  public constructTemplateOptions(data: DirFileNameSelection) {
-    const templateOptions: ApexClassOptions = {
-      template: 'DefaultApexClass',
-      classname: data.fileName,
-      outputdir: data.outputdir
-    };
-    return templateOptions;
+let initialized = false;
+let fileNameGatherer: ParametersGatherer<any>;
+let outputDirGatherer: ParametersGatherer<any>;
+let metadataTypeGatherer: ParametersGatherer<any>;
+function getParamGatherers() {
+  if (!initialized) {
+    fileNameGatherer = new SelectFileName(APEX_CLASS_NAME_MAX_LENGTH);
+    outputDirGatherer = new SelectOutputDir(APEX_CLASS_DIRECTORY);
+    metadataTypeGatherer = new MetadataTypeGatherer(APEX_CLASS_TYPE);
+    initialized = true;
   }
+  return {
+    fileNameGatherer,
+    outputDirGatherer,
+    metadataTypeGatherer
+  };
 }
```

Comment:
I think it's good to only every create the one instance of these for the command handler (and didn't want to change functionality) but I wanted to avoid creating instances on loading of the module so I created the function to wrap the creation of the gatherers.